Feature/login api 구현

Daeho-Son opened the pull request:
# 추가 및 변경 사항
### 로그인 요청 시, 에러가 발생하는 경우
- 해당 userId가 존재하지 않는 경우 (에러코드 AUTH-01)
- 잘못된 password인 경우(에러코드 AUTH-02)

### 로그인 요청 시, 성공하는 경우
- body에 accessToken(JWS) 반환

### ErrorCode의 code를 Integer에서 String으로 변경
- 에러의 코드를 AUTH-01, AUTH-02와 같이 정의해서 어떤 에러인지 찾기 쉽게 하기 위해 변경
![image](https://github.com/mattaeng/Spring/assets/48791587/4ea06350-3e9f-4710-9baf-b4d67b58bad7)

Daeho-Son committed:
Feat: login API 구현

추가 및 변경 사항
- 로그인 요청 시, 에러가 발생하는 경우
: 해당 userId가 존재하지 않는 경우 (에러코드 AUTH-01)
: 잘못된 password인 경우(에러코드 AUTH-02)

- 로그인 성공 시, body에 accessToken(JWS) 반환

- ErrorCode의 code를 Integer에서 String으로 변경
: 에러의 코드를 AUTH-01, AUTH-02와 같이 정의해서 어떤 에러인지 찾기 쉽게
  하기 위해 변경

Daeho-Son committed:
Docs: 스웨거에 api 추가

- POST /api/v1/users
- POST /api/v1/login

Daeho-Son committed:
Comment: User API에서 해야할 목록(TODO) 추가

youngseobso commented on src/main/java/com/mattaeng/mattaengapi/controller/AuthController.java:
```diff
@@ -0,0 +1,28 @@
+
+	@Operation(summary = "로그인 API")
+	@PostMapping("/login")
+	public Api<LoginResponse> login(@RequestBody LoginRequest request) {
```
request -> loginRequest같이 하는게 더 가독성이 좋지 않을까 생각합니다

Daeho-Son replied:
수정 완

youngseobso commented on src/main/java/com/mattaeng/mattaengapi/security/jwt/JwtProvider.java:
```diff
@@ -0,0 +1,56 @@
+
+	private final JwtParser jwtParser;
+
+	private final SecretKey key;
```
key -> secretKey가 더 괜찮지 않을까 생각합니다

Daeho-Son replied:
수정 완

youngseobso commented on src/main/java/com/mattaeng/mattaengapi/service/AuthService.java:
```diff
@@ -0,0 +1,36 @@
+	private final JwtProvider jwtProvider;
+	private final BCryptConfig bCryptConfig;
+
+	public LoginResponse login(LoginRequest request) {
```
request -> loginRequest같이 하는게 더 가독성이 좋지 않을까 생각합니다

Daeho-Son replied:
수정 완

youngseobso commented on src/main/java/com/mattaeng/mattaengapi/service/UserService.java:
```diff
@@ -19,9 +19,10 @@ public class UserService {
 	private final UserRepository userRepository;
 
+	// TODO: (추가) 휴대폰 인증, 약관 동의
 	public CreateUserResponse createUser(CreateUserRequest request) {
```
request -> createUserRequest

Daeho-Son replied:
수정 완

youngseobso reviewed (commented):
good~

Daeho-Son committed:
Style: 변수명의 네이밍을 명확하게 변경

- LoginReqeust request -> LoginRequest loginRequest
- CreateUserRequest request -> CreateUserRequest createUserRequest
- SecretKey key -> SecretKey secretKey

youngseobso reviewed (approved):
good-
